### PR TITLE
feat(backup): include timestamps in backup status

### LINKED ISSUE #10238: Backup status contains timestamps

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This updates `BackupStatus` to contain two timestamps, `created` and `lastModified`.

This also includes a somewhat larger refactoring to merge our two management objects, 'status.json' and 'metadata.json' so that only one object, 'manifest.json', needs to be queried and updated.

closes #10238

### CODE REVIEW

File: backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java [hunk 1]
```diff
@@ -156,25 +154,27 @@ public CompletableFuture<Backup> restore(final BackupIdentifier id, final Path t
...
   }
 
   @Override
   public CompletableFuture<BackupStatusCode> markFailed(
       final BackupIdentifier id, final String failureReason) {
     LOG.info("Marking {} as failed", id);
-    return setStatus(id, new Status(BackupStatusCode.FAILED, Optional.of(failureReason)));
+    return readManifestObject(id)
+        .thenComposeAsync(manifest -> writeManifestObject(manifest.withFailedStatus(failureReason)))
+        .thenApply(Manifest::statusCode);
```
Comment (deepthidevaki) [inline backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java:177]: :x: If manifest does not exist, or if there is an error reading it then we cannot mark it as failed. This is not ideal. Backup can be marked as failure even before writing the manifest. For example, when no valid snapshots are found, BackupManager will mark this backup as failed even before writing any objects to the backup store.

That case must be handled here - if there is no manifest, it should create an incomplete manifest with known information and set the status to fail. Then we must verify that we can read from an incomplete manifest without any issues.

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
Thanks @oleschoenburg . I'm not fully convinced if having one manifest with the metadata and the status is a good idea. But keeping them separate would also make it difficult to update the status and the timestamp atomically. I assume this is why you merged them. 
I'm ok with keeping them merged, but please see my comment and address those issues.

File: backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java (hunk 1 above)
Comment (oleschoenburg) [reply backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java]: Thanks for pointing this out, I completely missed this :+1: 

I've included (another) refactoring that allows for type checked manifests, more details in the commit message: 5b0a7730f2f0355febb538f7b32ba7c8ddeb2e22

There, I also fixed this issue and added a test here: cba12bb185f9148a1e075f05ddf679bb5b7ab9a7

**Review by deepthidevaki — APPROVED** [review]
Lots of changes for a small thing, but I like it :smile: :rocket: Please fix the alerts before merging.

**Pull request merged**